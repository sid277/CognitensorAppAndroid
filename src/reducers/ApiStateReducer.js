export const apiStateReducer = (state, action) => {
  switch (action.type) {
    case 'API_FETCH_INIT':
      return {
        ...state,
        isLoading: true,
        isError: false,
      };
    case 'API_FETCH_SUCCESS':
      return {
        ...state,
        isLoading: false,
        isError: false,
        data: action.payload,
      };
    case 'API_FETCH_FAILURE':
      return {
        ...state,
        isLoading: false,
        isError: true,
      };
    default:
      throw new Error();
  }
};
