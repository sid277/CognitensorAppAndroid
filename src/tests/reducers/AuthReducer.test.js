import React from 'react';
import { shallow } from 'enzyme';
import {authReducer} from '../../reducers/AuthReducer';
import renderer from 'react-test-renderer';


test('should test authReducer component', () => {
const wrapper = authReducer(undefined, {});
expect(wrapper).toMatchSnapshot();
});

test('should test authReducer for RESTORE_TOKEN action type', () => {
    const state = {isLoading: true,isSignout: false,userToken: null,}
    const action = {type: "RESTORE_TOKEN",token: "test_token"}
    const wrapper = authReducer(state, action );
    expect(wrapper).toEqual({...state,userToken: action.token,isLoading: false,});
    });
    

test('should test authReducer for SIGN_IN action type', () => {
    const state = {isLoading: true,isSignout: false,userToken: null,}
    const action = {type: "SIGN_IN",token: "test_token"}
    const wrapper = authReducer(state, action );
    expect(wrapper).toEqual({...state,isSignout: false,userToken: action.token,});
    });

test('should test authReducer for SIGN_OUT action type', () => {
    const state = {isLoading: true,isSignout: false,userToken: null,}
    const action = {type: "SIGN_OUT",token: "test_token"}
    const wrapper = authReducer(state, action );
    expect(wrapper).toEqual({...state,isSignout: true,userToken: null,});
    });