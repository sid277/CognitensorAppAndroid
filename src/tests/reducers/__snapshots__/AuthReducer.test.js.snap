// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test authReducer component 1`] = `undefined`;
